bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,19 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://'  # http:// or https://
+            r'[^.:%\s_/?#[\]@\\]+'  # Initial Part of Domain
+            r'\.'  # A required dot (.)
+            r'(?:[^\s()%\'\"`<>|\\]+'  # Domain / Subdomain Name
+            r'|'  # OR
+            r'\([^\s()%\'\"`<>|\\]*\))'
+            # Subdomain Name that is contained within ()
+            r'%?[0-9]*'  # A % sign (0 or 1 times) followed by any number
```

Comment:
So, if no letters are allowed, are we assuming a google search like "cpp %s error" is not a valid link that we will check ?
Example: https://encrypted.google.com/search?hl=en&q=cpp+%s+error

If this is expected, can this be added as a test case with a comment stating this ?